### PR TITLE
feat: Adding demo project

### PR DESCRIPTION
**Pull request by benmckillop**
# feat: Adding demo project

_Thank you for your contribution to the project._

Adding a demo project so that we can run unit tests that test tapping buttons etc.

# Checklist

## Before raising your pull request:
- [X] Ran the app locally ensuring it builds 
- [X] Ran the tests locally ensuring they pass on Build
- [X] Pull request has a clear title with ticket ID and a short description about the feature or update
      i.e. _DCMAW-222: Added ReadID SDK to iOS app_
- [X] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [X] Met all of the acceptance criteria specified in the user story on Jira
- [X] Reviewed your own code to ensure you are following the style guidelines
- [X] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [X] Written Unit and Integration tests if needed

- [ ] ~Met all accessibility requirements?~
    - [ ] ~Checked dynamic type sizes are applied~
    - [ ] ~Checked VoiceOver can navigate your new code~
    - [ ] ~Checked a user can navigate only using a keyboard around your new code~

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.
- [ ] Targeted the correct branch; `develop`, `release` or `main`

### CODE REVIEW

File: GDSCommon-Demo/GDSCommon-Demo/SceneDelegate.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
+//
+//  SceneDelegate.swift
+//  GDSCommon-Demo
+//
+//  Created by McKillop, Ben on 21/09/2023.
+//
+
```
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/SceneDelegate.swift:7]: Please could we remove file headers? 🙂

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
             
       - name: Build and Test
         run: |
-          xcodebuild -scheme GDSCommon-Package test -destination "platform=iOS Simulator,name=iPhone 14,OS=latest" \
+          xcodebuild -scheme GDSCommon-Demo test -destination "platform=iOS Simulator,name=iPhone 14,OS=latest" \
```
Comment (benmckillop) [inline .github/workflows/pull-request.yml:34]: @amopatel1234 Is this change needed, or do we need both here?

**Comment by benmckillop** [comment]
Uploading video.mov…

File: .github/workflows/pull-request.yml (hunk 2 above)
Comment (amopatel1234) [reply .github/workflows/pull-request.yml:34]: if the `GDSCommon-Demo` scheme contains the tests for the package then no both are not needed, just demo is fine

File: GDSCommon-Demo/GDSCommon-Demo/Base.lproj/Main.storyboard
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Base.lproj/Main.storyboard:1]: Appreciate this is a demo, but I think we should set this up naming the storyboard and view controllers following our naming conventions to describe what they do.

File: GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json:1]: Wonder if it is worth setting an accent colour so that the elements that are styled by the accent colour look more like they are intended?

File: GDSCommon-Demo/GDSCommon-Demo/Base.lproj/Main.storyboard
Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-Demo/Base.lproj/Main.storyboard:1]: Yes good idea, thank you :)

File: GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json
Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json:1]: Yes good idea, thank you :)

**Review by robinphillips — APPROVED** [review]

**Review by robinphillips — COMMENTED** [review]
Should we add some documentation to the package readme for this demo project to explain how to use it?

File: GDSCommon-Demo/GDSCommon-DemoTests/GDSCommon_DemoTests.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//
+//  GDSCommon_DemoTests.swift
+//  GDSCommon-DemoTests
+//
+//  Created by McKillop, Ben on 21/09/2023.
+//
+
```
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-DemoTests/GDSCommon_DemoTests.swift:7]: Please could we remove the file header?

File: GDSCommon-Demo/GDSCommon-Demo/Mocks/Screens.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import GDSCommon
+import UIKit
+
+enum Screens: CaseIterable {
```
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Mocks/Screens.swift:4]: Is it worth adding some DocC here?

File: .swiftlint.yml [hunk 5]
```diff
@@ -4,6 +4,8 @@ excluded:
 - Frameworks
 - Dependencies
 - SnapshotHelper.swift
+- GeneratedAssetSymbols.swift
+- resource_bundle_accessor.swift
 - MockReadID
```
Comment (robinphillips) [inline .swiftlint.yml:9]: Should remove `MockReadID` from here

File: GDSCommon-Demo/GDSCommon-Demo/Mocks/Screens.swift (hunk 4 above)
Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-Demo/Mocks/Screens.swift:14]: Done!

File: GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+          "value" : "dark"
+        }
+      ],
+      "color" : {
+        "color-space" : "srgb",
+        "components" : {
+          "alpha" : "1.000",
+          "blue" : "0.373",
+          "green" : "0.702",
+          "red" : "0.000"
```
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json:28]: Should this be a hex colour?

File: GDSCommon-Demo/GDSCommon-Demo/Mocks/MockInstructionWithImageViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import GDSCommon
+import UIKit
+
+class MockInstructionWithImageViewModel: InstructionsWithImageViewModel {
+    var title: GDSCommon.GDSLocalisedString
```
Comment (robinphillips) [inline GDSCommon-Demo/GDSCommon-Demo/Mocks/MockInstructionWithImageViewModel.swift:5]: I don't think we need to specifically reference `GDSCommon` namespace on these types? I don't think it is _wrong_ just perhaps more wordy than it needs to be? What do you think? Same would apply for the `ButtonViewModel`

Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-Demo/Mocks/MockInstructionWithImageViewModel.swift:5]: Yeah it's a fair point. I had thought that it was a little more explanatory but when I looked again there was a lot 😅 going to remove. Thank you!

File: GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json (hunk 6 above)
Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-Demo/Assets.xcassets/AccentColor.colorset/Contents.json]: Fixed now!

**Review by robinphillips — APPROVED** [review]
Great job! This is going to be really helpful 🥳

File: GDSCommon-Demo/GDSCommon-DemoTests/GDSCommon_DemoTests.swift (hunk 3 above)
Comment (benmckillop) [reply GDSCommon-Demo/GDSCommon-DemoTests/GDSCommon_DemoTests.swift]: Done!

**Review by robinphillips — APPROVED** [review]
